subsys/net/lib/nrf_cloud/src/nrf_cloud_transport.c

```diff
@@ -331,6 +335,76 @@ static int nct_client_id_set(const char * const client_id)
 	return -ENOTRECOVERABLE;
 }
 
+/* Function to get the client id */
+int nct_client_id_get(char *id, size_t id_len)
+{
+	int len = strlen(client_id_buf);
+
+	if (id_len <= len) {
+		return -EMSGSIZE;
+	}
+	if (client_id_buf && (id != NULL) && len) {
```

Comment:
Perhaps use `else if` here.

Reply:
I think I will get an "unnecessary else" warning from the compliance checker, but I'll try it.  I know I've had to remove some 'else's before even though it seems natural to have them.

Reply:
Must be for a slightly different situation.  Done.